rotkehlchen/tests/unit/decoders/test_curve.py

```diff
@@ -808,3 +809,149 @@ def test_deposit_multiple_tokens(ethereum_transaction_decoder, ethereum_accounts
         ),
     ]
     assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0xdE206bC0Fde2eF5C8BB6A1d552a64F82A2407Be4']])
+def test_curve_usdn_add_liquidity(ethereum_transaction_decoder, ethereum_accounts):
+    """Check that adding liquidity to a curve pool using the USDN contract is properly decoded."""
+    tx_hex = deserialize_evm_tx_hash('0x6c28df56ae4a7f784577273f72402a9b6640024327ee952fdde72c9cfdf08da5')  # noqa: E501
+    evmhash = deserialize_evm_tx_hash(tx_hex)
+    user_address = ethereum_accounts[0]
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_transaction_decoder.evm_inquirer,
+        database=ethereum_transaction_decoder.database,
+        tx_hash=tx_hex,
+    )
+    expected_events = [
+        HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=0,
+            timestamp=1674470159000,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(FVal('0.005672980418415474')),
+            location_label=user_address,
+            counterparty=CPT_GAS,
+            notes='Burned 0.005672980418415474 ETH for gas',
+        ), HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=78,
+            timestamp=1674470159000,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=Asset('eip155:1/erc20:0x4f3E8F405CF5aFC05D68142F3783bDfE13811522'),
+            balance=Balance(FVal('2676.003172633078929284')),
+            location_label=user_address,
+            counterparty=CPT_CURVE,
+            notes='Receive 2676.003172633078929284 usdn3CRV after depositing in curve pool',
+            extra_data={'deposit_events_num': 1},
+        ), HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=85,
+            timestamp=1674470159000,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_USDC,
+            balance=Balance(FVal('761.396655')),
+            location_label=user_address,
+            counterparty=CPT_CURVE,
+            notes='Deposit 761.396655 USDC in curve pool 0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7',  # noqa: E501
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0x6d84264A7bD2Cffa4A117BA2350403b3A9866949']])
+def test_curve_usdn_remove_liquidity(ethereum_transaction_decoder, ethereum_accounts):
+    """Check that removing liquidity from a curve pool using the USDN contract is properly decoded."""  # noqa: E501
```

Comment:
Are there tests for other zaps?

Reply:
there's an existing test for curve y deposit(a zap)